base/compiler/ssair/show.jl

```diff
@@ -66,6 +66,26 @@ function print_stmt(io::IO, idx::Int, @nospecialize(stmt), used::BitSet, maxleng
         end
         join(io, (print_arg(i) for i = 3:length(stmt.args)), ", ")
         print(io, ")")
+    elseif isexpr(stmt, :call) && length(stmt.args) >= 1
+        arg1 = stmt.args[1]
+        if arg1 isa GlobalRef && isdefined(arg1.mod, arg1.name)
+            arg1 = getfield(arg1.mod, arg1.name)
+        end
```

Comment:
It would be probably better to use `ft = Core.Compiler.argextype(arg1, src); f = Core.Compiler.singleton_type(ft)`?

Reply:
Yeah, that's a good point - there are already a couple of subtleties I forgot about here w.r.t. binding lookups

Thanks!

Reply:
Hmm, so the challenge is that the `CodeInfo` has actually forgotten its `sptypes` by the time we print it: https://github.com/JuliaLang/julia/blob/4cdd864e535b16e928c2eff43e6f1583bd77209d/base/show.jl#L2859

Any idea how we can recover those?

Reply:
Those are now guaranteed to be at `.parent.def.sparam_vals` for any valid CodeInfo (this code may have to occasionally print malformed CodeInfo, so use some care when accessing that however)

Reply:
Great, thanks for the tip!